pages/api/users/[username]/links/[url].js

```diff
@@ -0,0 +1,121 @@
+import Link from "../../../../../models/Link";
+import Profile from "../../../../../models/Profile";
+import Stats from "../../../../../models/Stats";
+import connectMongo from "../../../../../config/mongo";
+
+/**
+ * @type {import('next').NextApiHandler}
+ */
+export default async function handler(req, res) {
+  await connectMongo();
+
+  const { username, url } = req.query;
+
+  if (req.method != "GET" || !username || !url) {
+    return res
+      .status(400)
+      .json({ error: "Invalid request: username and link required" });
+  }
+
+  let getProfile = await Profile.findOne({ username });
+
+  if (!getProfile) {
+    try {
+      await Profile.create({
+        username,
+        views: 1,
+      });
+
+      getProfile = await Profile.findOne({ username });
+    } catch (e) {
+      const err = `ERROR creating profile: ${e}`;
+      console.log(err);
+      return res.status(500).json({ message: err });
+    }
+  }
+
+  const getLink = await Link.findOne({ username, url });
+
+  if (getLink) {
+    try {
+      await Link.updateOne(
+        {
+          username,
+          url,
+        },
+        {
+          $inc: { clicks: 1 },
+        }
+      );
+    } catch (e) {
+      const err = `ERROR incrementing link stats: ${e}`;
+      console.log(err);
+      return res.res(500).json({ message: err });
+    }
+  }
+
+  if (!getLink) {
+    try {
+      const link = await Link.create({
+        profile: getProfile._id,
+        username,
+        url,
+        clicks: 1,
+      });
+
+      await Profile.updateOne(
+        {
+          username,
+        },
+        {
+          $push: { links: link._id },
+        },
+        { new: true, useFindAndModify: false }
+      );
+    } catch (e) {
+      const err = `ERROR creating link: ${e}`;
+      console.log(err);
+      return res.res(500).json({ message: err });
+    }
+  }
+
+  const date = new Date();
+
+  date.setHours(1, 0, 0, 0);
+
+  const getPlatformStats = await Stats.findOne({ date });
+
+  if (getPlatformStats) {
+    try {
+      await Stats.updateOne(
+        {
+          date,
+        },
+        {
+          $inc: { clicks: 1 },
+        }
+      );
+    } catch (e) {
+      const err = `ERROR incrementing platform stats: ${e}`;
+      console.log(err);
+      return res.res(500).json({ message: err });
+    }
+  }
+
+  if (!getPlatformStats) {
+    try {
+      await Stats.create({
+        date,
+        views: 0,
+        clicks: 1,
+        users: 0,
+      });
+    } catch (e) {
+      const err = `ERROR incrementing platform stats: ${e}`;
+      console.log(err);
+      return res.res(500).json({ message: err });
+    }
+  }
+
+  return res.status(201).redirect(decodeURIComponent(url));
+}
```

Comment:
There are only a few lines difference with the file in statistics, less changes are more 😉 

It would be better to add the few tweaks to the original file and move it, rather than creating a new duplicate file 
https://github.com/EddieHubCommunity/LinkFree/blob/main/pages/api/statistics/%5B...data%5D.js

Reply:
No no no. I've done exactly that I just forgot to push deleting the statistics file, did that now

Reply:
I've improved that file itself

Reply:
@krshkun I did a diff there are only a few line changes please keep it simple...

```diff
diff --git a/pages/api/statistics/[...data].js b/pages/api/statistics/[...data].js
index 7e05e900..47c0d572 100644
--- a/pages/api/statistics/[...data].js
+++ b/pages/api/statistics/[...data].js

 export default async function handler(req, res) {
   await connectMongo();
-  const { data } = req.query;
-  const username = data[0];
-  const url = data[1];
 
-  if (req.method != "PUT" || !username || !url) {
+  const { username, url } = req.query;
+
+  if (req.method != "GET" || !username || !url) {
     return res
       .status(400)
-      .json({ error: "Invalid request: 'username' and url 'required'" });
+      .json({ error: "Invalid request: username and link required" });
   }
 
-  const getProfile = await Profile.findOne({ username });
+  let getProfile = await Profile.findOne({ username });
 
   if (!getProfile) {
-    return res
-      .status(404)
-      .json({ error: "Invalid request: 'username' not found" });
+    try {
+      await Profile.create({
+        username,
+        views: 1,
+      });
+
+      getProfile = await Profile.findOne({ username });
+    } catch (e) {
+      const err = `ERROR creating profile: ${e}`;
+      console.log(err);
+      return res.status(500).json({ message: err });
+    }
   }
 
   const getLink = await Link.findOne({ username, url });
+
   if (getLink) {
     try {
       await Link.updateOne(
@@ -36,7 +48,9 @@ export default async function handler(req, res) {
         }
       );
     } catch (e) {
-      console.log("ERROR incrementing link", e);
+      const err = `ERROR incrementing link stats: ${e}`;
+      console.log(err);
+      return res.res(500).json({ message: err });
     }
   }
 
@@ -48,6 +62,7 @@ export default async function handler(req, res) {
         url,
         clicks: 1,
       });
+
       await Profile.updateOne(
         {
           username,
@@ -58,13 +73,18 @@ export default async function handler(req, res) {
         { new: true, useFindAndModify: false }
       );
     } catch (e) {
-      console.log("ERROR creating link", e);
+      const err = `ERROR creating link: ${e}`;
+      console.log(err);
+      return res.res(500).json({ message: err });
     }
   }
 
   const date = new Date();
+
   date.setHours(1, 0, 0, 0);
+
   const getPlatformStats = await Stats.findOne({ date });
+
   if (getPlatformStats) {
     try {
       await Stats.updateOne(
@@ -76,7 +96,9 @@ export default async function handler(req, res) {
         }
       );
     } catch (e) {
-      console.log("ERROR incrementing platform stats", e);
+      const err = `ERROR incrementing platform stats: ${e}`;
+      console.log(err);
+      return res.res(500).json({ message: err });
     }
   }
 
@@ -89,10 +111,11 @@ export default async function handler(req, res) {
         users: 0,
       });
     } catch (e) {
-      console.log("ERROR creating platform stats", e);
+      const err = `ERROR incrementing platform stats: ${e}`;
+      console.log(err);
+      return res.res(500).json({ message: err });
     }
   }
 
-  const latestLink = await Link.findOne({ username, url });
-  res.status(201).json(latestLink);
+  return res.status(201).redirect(decodeURIComponent(url));
 }
```

Then we can see the exact changes and discuss the specific line changes more easily

Reply:
I'm not really sure what do you want me to do in the PR regarding the API requests, every time, even if I revert everything, just have the [...data].js file on the upstream and move it to my directory, it shows them as two separate changes. If you want to try this out, just open `EddieHubCommunity/LinkFree:main` and change the file's name and commit or something

Reply:
This is the diff

```diff
1,5c1,8
--- pages/api/statistics/[...data].js	2023-01-14 21:18:09.523506221 +0530
+++ pages/api/users/[username]/links/[url].js	2023-01-14 21:14:51.613508229 +0530
@@ -1,29 +1,41 @@
-import Link from "../../../models/Link";
-import Profile from "../../../models/Profile";
-import Stats from "../../../models/Stats";
-import connectMongo from "../../../config/mongo";
-
+import Link from "../../../../../models/Link";
+import Profile from "../../../../../models/Profile";
+import Stats from "../../../../../models/Stats";
+import connectMongo from "../../../../../config/mongo";
+
+/**
+ * @type {import('next').NextApiHandler}
+ */
 export default async function handler(req, res) {
   await connectMongo();
-  const { data } = req.query;
-  const username = data[0];
-  const url = data[1];
 
-  if (req.method != "PUT" || !username || !url) {
+  const { username, url } = req.query;
+
+  if (req.method != "GET" || !username || !url) {
     return res
       .status(400)
-      .json({ error: "Invalid request: 'username' and url 'required'" });
+      .json({ error: "Invalid request: username and link required" });
   }
 
-  const getProfile = await Profile.findOne({ username });
+  let getProfile = await Profile.findOne({ username });
 
   if (!getProfile) {
-    return res
-      .status(404)
-      .json({ error: "Invalid request: 'username' not found" });
+    try {
+      await Profile.create({
+        username,
+        views: 1,
+      });
+
+      getProfile = await Profile.findOne({ username });
+    } catch (e) {
+      const err = `ERROR creating profile: ${e}`;
+      console.log(err);
+      return res.status(500).json({ message: err });
+    }
   }
 
   const getLink = await Link.findOne({ username, url });
+
   if (getLink) {
     try {
       await Link.updateOne(
@@ -36,7 +48,9 @@
         }
       );
     } catch (e) {
-      console.log("ERROR incrementing link", e);
+      const err = `ERROR incrementing link stats: ${e}`;
+      console.log(err);
+      return res.res(500).json({ message: err });
     }
   }
 
@@ -48,6 +62,7 @@
         url,
         clicks: 1,
       });
+
       await Profile.updateOne(
         {
           username,
@@ -58,13 +73,18 @@
         { new: true, useFindAndModify: false }
       );
     } catch (e) {
-      console.log("ERROR creating link", e);
+      const err = `ERROR creating link: ${e}`;
+      console.log(err);
+      return res.res(500).json({ message: err });
     }
   }
 
   const date = new Date();
+
   date.setHours(1, 0, 0, 0);
+
   const getPlatformStats = await Stats.findOne({ date });
+
   if (getPlatformStats) {
     try {
       await Stats.updateOne(
@@ -76,7 +96,9 @@
         }
       );
     } catch (e) {
-      console.log("ERROR incrementing platform stats", e);
+      const err = `ERROR incrementing platform stats: ${e}`;
+      console.log(err);
+      return res.res(500).json({ message: err });
     }
   }
 
@@ -89,10 +111,11 @@
         users: 0,
       });
     } catch (e) {
-      console.log("ERROR creating platform stats", e);
+      const err = `ERROR incrementing platform stats: ${e}`;
+      console.log(err);
+      return res.res(500).json({ message: err });
     }
   }
 
-  const latestLink = await Link.findOne({ username, url });
-  res.status(201).json(latestLink);
+  return res.status(201).redirect(decodeURIComponent(url));
 }
```

Reply:
Not sure I follow what the issue is, you shared the the same diff as I shared above it.

We don't need both files as they are almost identical, so we don't lose the history, move the file over to the new location with your changes on top - then we can discuss the changes

Reply:
@eddiejaoude when I move the files, it shows me 2 different changes, 1 for creating a new file and another for deleting